### PR TITLE
fix demo app

### PR DESCRIPTION
**Pull request by trin4ik**
-- fix require `expo-router/metro-config` to `expo/metro-config`
-- add `config.transformer.unstable_allowRequireContext = true` to metro config
-- fix `Open Modal` link
-- fix `<Link href="../"` to `<Link to="../"`
-- fix `Open settings dynamic` in modal link
-- `<span>` to `<Text>` component in modal

and my phpStorm reformat code by my template (( if it is important, I can remake it with your format
tested with expo go on ios

### CODE REVIEW

File: apps/demo/metro.config.js [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
-  path.resolve(projectRoot, "node_modules"),
-  path.resolve(workspaceRoot, "node_modules"),
-];
+	path.resolve(projectRoot, "node_modules"),
+	path.resolve(workspaceRoot, "node_modules"),
+]
 // 3. Force Metro to resolve (sub)dependencies only from the `nodeModulesPaths`
-config.resolver.disableHierarchicalLookup = true;
+config.resolver.disableHierarchicalLookup = true
+config.transformer.unstable_allowRequireContext = true
```
Comment (EvanBacon) [inline apps/demo/metro.config.js:21]:
```suggestion
```
Not needed anymore

**Review by EvanBacon — CHANGES_REQUESTED** [review]
fix the formatting please.

**Comment by trin4ik** [comment]
> fix the formatting please.

i request new pull
https://github.com/expo/router/pull/30

File: apps/demo/metro.config.js (hunk 1 above)
Comment (trin4ik) [reply apps/demo/metro.config.js:21]: in my config, it need
```bash
$ uname -a
Linux StayHome 5.15.0-47-generic #51-Ubuntu SMP Thu Aug 11 07:51:15 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

$ npm -v
8.19.2

$ yarn -v
1.22.18

$ node -v
v16.17.0
```

**Pull request closed without merging**